src/App.js

```diff
@@ -1,21 +1,58 @@
-import React, { Component } from "react";
+import React from "react";
 import { Route, BrowserRouter, Switch } from "react-router-dom";
 import DashboardPage from "./pages/DashboardPage";
 import SignIn from "./modules/sign/components/Signin";
+import { createContext } from "react"
+import { useReducer} from "react"
+
+const reducer = (state, { type, payload }) => {
+  switch (type) {
+    case "LOGIN":
+      return { ...state, loading: true };
+
+    case "LOGIN_SUCESS":
+      return { ...state, user: payload, loading: false };
+
+    case "LOGIN_FAILED":
+      return { ...state, user: {}, error: payload, loading: false };
+
+    case "GITLAB_AUTH":
+      localStorage.setItem('auth', "gitlab")
+      return { ...state, isGitlabAuth: true }
+    
+    case "CLEAR_GITLAB_AUTH":
+      localStorage.setItem('auth', "github")
+      return { ...state, isGitlabAuth: false }
+    default:
+      return state
+  }
+};
+
+const initialState = {
+  loggedIn: localStorage.getItem("loggedIn"),
```

Comment:
I don't understand why there is a separate `loggedIn` state and also a user object. Any reason for this?

Reply:
The whole workflow was implemented by @devCayacoa, I just built the implementation on top of it. I have no idea why there's a user object. I'll refactor the codes as expected. Thanks.